mergify_engine/web/api/queues.py

```diff
@@ -317,5 +325,367 @@ async def repository_queues(
+    if reason == "":
+        reason = "No freeze reason was specified."
+
+    async with github.aget_client(installation_json) as client:
+        try:
+            # Check this token as access to this repository
+            repo = typing.cast(
+                github_types.GitHubRepository,
+                await client.item(f"/repos/{owner}/{repository}"),
+            )
+        except (http.HTTPNotFound, http.HTTPForbidden, http.HTTPUnauthorized):
+            raise fastapi.HTTPException(status_code=404)
+
+        sub = await subscription.Subscription.get_subscription(
+            redis_cache, repo["owner"]["id"]
+        )
+        installation = context.Installation(
+            installation_json, sub, client, redis_cache, redis_queue
+        )
+        repository_ctxt = installation.get_repository_from_github_data(repo)
+        config_file = await repository_ctxt.get_mergify_config_file()
+
+        if config_file is not None:
+            config = get_mergify_config(config_file)
+            queue_rules = config["queue_rules"]
+
+            if all(queue_name != rule.name for rule in queue_rules):
+                return responses.JSONResponse(
+                    status_code=400,
+                    content={"error": f'The queue "{queue_name}" does not exists'},
+                )
+
+            qf = await QueueFreeze.get(repository_ctxt, queue_name)
+            if qf is not None:
+                if qf.reason != reason:
+                    qf.reason = reason
+                    await qf.save()
+                    return responses.JSONResponse(
+                        status_code=200,
+                        content={
+                            "updated_queue_freeze_data": qf.get_json(),
+                            "description": f'The queue "{queue_name}" was already frozen. Its data have been successfully updated',
+                        },
+                    )
+                else:
+                    return responses.JSONResponse(
+                        status_code=200,
+                        content={
+                            "description": f'The queue "{queue_name}" was already frozen. Nothing happened.'
+                        },
+                    )
+
+            qf = QueueFreeze(
+                repository=repository_ctxt,
+                name=queue_name,
+                reason=reason,
+                user_app_name=application.name,
+                freeze_date=date.utcnow(),
+                freeze_state=True,
+            )
+            await qf.save()
+            return responses.JSONResponse(
+                status_code=200,
+                content={
+                    "description": f'The queue "{queue_name}" has been successfully frozen'
+                },
+            )
+        else:
+            return responses.JSONResponse(
+                status_code=400,
+                content={
+                    "error": "Configuration file .mergify.yml is missing",
```

Comment:
```suggestion
                    "error": "Mergify configuration file is missing",
```

it could be in `.config/mergify.yml` or the like